pkg/component/server/konnectivity.go

```diff
@@ -73,65 +74,99 @@ func (k *Konnectivity) Init() error {
 		return fmt.Errorf("failed to chown %s: %v", k.K0sVars.KonnectivitySocketDir, err)
 	}
 
-	// set default serverCount to 1
-	k.serverCount = kubernetes.MinLeaseHolders
+	// Buffered chan to send updates for the count of servers
+	k.serverCountChan = make(chan int, 1)
+
+	k.stopCtx, k.stopFunc = context.WithCancel(context.Background())
```

Comment:
I think those should be moved to the beginning of `Run()`. otherwise they are not cleaned up if some of the other components' `Init()` fails and Stop() never called.